temporalio/client.py

```diff
@@ -5383,49 +6057,132 @@ async def start_workflow_update(
             temporalio.common._apply_headers(
                 input.headers, req.request.input.header.fields
             )
+        return req
 
-        # Repeatedly try to invoke start until the update reaches user-provided
-        # wait stage or is at least ACCEPTED (as of the time of this writing,
-        # the user cannot specify sooner than ACCEPTED)
-        resp: temporalio.api.workflowservice.v1.UpdateWorkflowExecutionResponse
-        while True:
-            try:
-                resp = await self._client.workflow_service.update_workflow_execution(
-                    req,
-                    retry=True,
-                    metadata=input.rpc_metadata,
-                    timeout=input.rpc_timeout,
-                )
-            except RPCError as err:
-                if (
-                    err.status == RPCStatusCode.DEADLINE_EXCEEDED
-                    or err.status == RPCStatusCode.CANCELLED
-                ):
-                    raise WorkflowUpdateRPCTimeoutOrCancelledError() from err
-                else:
-                    raise
-            except asyncio.CancelledError as err:
+    async def start_update_with_start_workflow(
+        self, input: StartWorkflowUpdateWithStartInput
+    ) -> WorkflowUpdateHandle[Any]:
+        seen_start = False
+
+        def on_start(
+            start_response: temporalio.api.workflowservice.v1.StartWorkflowExecutionResponse,
+        ):
+            nonlocal seen_start
+            if not seen_start:
+                input._on_start(start_response)
+                seen_start = True
+
+        err: Optional[BaseException] = None
+
+        try:
+            return await self._start_workflow_update_with_start(
+                input.start_workflow_input, input.update_workflow_input, on_start
+            )
+        except asyncio.CancelledError as _err:
+            err = _err
+            raise WorkflowUpdateRPCTimeoutOrCancelledError() from err
+        except RPCError as _err:
+            err = _err
+            if err.status in [
+                RPCStatusCode.DEADLINE_EXCEEDED,
+                RPCStatusCode.CANCELLED,
+            ]:
                 raise WorkflowUpdateRPCTimeoutOrCancelledError() from err
+            else:
+                multiop_failure = (
+                    temporalio.api.errordetails.v1.MultiOperationExecutionFailure()
+                )
+                if err.grpc_status.details[0].Unpack(multiop_failure):
+                    status = next(
+                        (
+                            st
+                            for st in multiop_failure.statuses
+                            if (
+                                st.details
```

Comment:
I think this logic is a bit off. I don't think we should _require_ details to be considered an error

Reply:
Thanks, that sounds right. 5a06a6a1efe869e79894787923f59da91ec1618f